src/core/lib/iomgr/exec_ctx.cc

```diff
@@ -99,11 +105,19 @@ gpr_timespec grpc_millis_to_timespec(grpc_millis millis,
 }
 
 grpc_millis grpc_timespec_to_millis_round_down(gpr_timespec ts) {
+  if (ts.clock_type == GPR_TIMESPAN) {
+    // TODO(soheil): specialize for infitinte past and future.
+    return timespan_to_millis_round_down(gpr_time_add(g_start_time, ts));
```

Comment:
Why do we need to add `ts` to `g_start_time`? If `ts` is `GPR_TIMESPAN`, we can use that directly to calculate millis?

Reply:
`grpc_timespec_to_millis_round_down` and `grpc_timespec_to_millis_round_up` both return the millis from `g_start_time`. So they are not msec duration, but the msec passed since  `g_start_time`. For example, `Now()` is a direct call to this method.

I admit this code path is utterly confusing.